File: app-java/src/main/java/science/monke/GreetingResource.java
package science.monke;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/greeting")
public class GreetingResource {

  @Inject
  GreetingConfig greetingConfig;

  @GET
  @Produces(MediaType.TEXT_PLAIN)
  public String hello() {
    return greetingConfig.message();
  }
}

File: app-kotlin/src/main/kotlin/org/acme/rest/GreetingResource.kt
package org.acme.rest

import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/greeting")
class GreetingResource {

    @Inject
    lateinit var greetingConfig: GreetingConfig

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = greetingConfig.message()
}
